packages/mdc-theme/_functions.scss

```diff
@@ -51,3 +56,66 @@
     @return "light";
   }
 }
+
+// Return the given number if $value >= $min and $value <= $max. If $value falls
+// outside those bounds, return the nearest boundary value ($max or $min).
+@function mdc-theme-clamp-number_($value, $min, $max) {
+  @return max($min, min($max, $value));
+}
+
+// lighten() and darken() require values to be between 0% and 100%.
+@function mdc-theme-clamp-lighten-amount_($percentage) {
+  @return mdc-theme-clamp-number_($percentage, 0%, 100%);
+}
+
+// Darken the given color by approximately the specified number of indexes
+// within its tonal palette.
+//
+// If the color is already very dark, it will be lightened instead of darkened
+// to ensure that the returned value is visually distinct from the input color.
+//
+// If the color is very light, it will be darkened twice as much as usual to
+// ensure that the returned value is visually distinct from the light variant
+// (which will actually end up being a dark shade).
+@function mdc-theme-dark-variant($color, $num-indexes: 2) {
+  $offset: $_mdc-theme-tonal-offset;
```

Comment:
Is this a style guide convention? Why not just reference `$_mdc-theme-tonal-offset` on the next line?